src/bloqade/squin/rewrite/wrap_analysis.py

```diff
@@ -70,3 +60,34 @@ def rewrite_Statement(self, node: ir.Statement) -> RewriteResult:
             if self.wrap(result):
                 has_done_something = True
         return RewriteResult(has_done_something=has_done_something)
```

Comment:
```suggestion
        return RewriteResult(has_done_something=any(self.wrap(result) for result in node.results)
```
could be a one-liner